src/training/train.py

```diff
@@ -94,17 +97,33 @@ def train_one_epoch(model, data, epoch, optimizer, scaler, scheduler, args, tb_w
 
         if args.accum_freq == 1:
             with autocast():
-                image_features, text_features, logit_scale = model(images, texts)
-                total_loss = loss(image_features, text_features, logit_scale)
+                model_out = model(images, texts)
+                # for clip if it does not output_dict
+                module = model.module if type(model) == DistributedDataParallel else model
+                if is_clip(module) and not module.output_dict:
+                    model_out = postprocess_clip_output(model_out)
```

Comment:
that last thing of note in my list, I don't feel there is any need to supoort clip not outping a dict
* allowing both output modes in the model was that so users in other training codebases, or in existing colab notebooks, etc wouldn't have full break in their code when they update open_clip
* our training code can always operate in outpu_dict mode, I don't see a need to support both
* if anyone is using open_clip train code with their own models, they can adapt it easily enough (they'll be more advanced users)

Reply:
makes sense, can you create an issue for it ?